api/views/api.py

```diff
@@ -11,15 +12,23 @@ def get_data(request):
         if request is None:
             raise Exception("Houston, we have a problem.")
 
-        search = request.GET.get("search")
+        # search = request.GET.get("search")
         page = request.GET.get("page")
         per_page = request.GET.get("per_page")
 
-        print("options", search, page, per_page)
-
         products = Product.objects.fully_loaded_objects()
+        products_for_serialization = products
+
+        if page is not None and per_page is not None:
+            paginator = Paginator(products, per_page)
+            products_for_serialization = paginator.get_page(page)
+
+            if len(products) / int(per_page) < int(page) or int(page) == 0:
+                products_for_serialization = []
```

Comment:
if we have 55 products with per page of 10, (pg1: 1-10, pg2: 11-20, pg3: 21-30, pg4: 31-40, pg5: 41-50, pg6: 51-55)

if I'm reading this right
based on this logic - len(products) for page 6 would be 5; 5 / per_page(10) would be 0; 0 < 6  - now we won't show the results? I would want to see partial pages.